utrox/sketchbook
# Add and update tests (Issue #9)

Add tests for the following operations and cases:

- Test query me
    - Authenticated user can query data about themselves
    - Unauthenticated users get no data

- Test query userprofile:
    - Returns data about the queried user by username
    - Returns an error message if querying nonexistant user

- Test post history for user:
    - Test pagination for post history
    - Test that only correct posts show up
    - Test that all posts show up which should show up
    - Test post history with zero posts
    - Test post history for nonexistant user

- Updating user profile:
    - Non-authenticated users recieve an error message if they try to update their 'profile'
    - Test ability for authenticated users to:
        - change their username
        - change their password - change their profile picture - change their profile background image

    - Tests for username validation, and appropriate error message: - Minimum amount of characters is 3 - Maximum amount of characters is 20 - Usernames must be unique - Usernames must be alphanumerical

    - Tests for password validation, and appropriate error message:
        - Passwords should be at least 8 characters long.
        - Passwords should be a maximum of 32 characters long. - Passwords should not be only numeric. - Password should not be too common.

    - Tests for image validation: - Images must have an accepted file extension (.jpg, .png, .webp) - Images must be at least 100x100px in size. - Images must be a maximum of 2000x2000px in size. - Images must be under 5mb.
